src/core/lib/resolver/resolver_registry.cc

```diff
@@ -24,20 +24,21 @@
 
 #include <vector>
 
+#include "absl/memory/memory.h"
 #include "absl/strings/str_cat.h"
 #include "absl/strings/str_format.h"
 
 #include <grpc/support/alloc.h>
 #include <grpc/support/log.h>
 #include <grpc/support/string_util.h>
 
-namespace grpc_core {
+#include "src/core/lib/resolver/resolver_registry.h"
 
-namespace {
+namespace grpc_core {
 
-class RegistryState {
+class ResolverRegistry::State {
```

Comment:
I sent you ctiller/grpc#451 to do some more cleanup here, which (amongst other things) removes the need for this class.  I think this is mostly a hold-over from before we could use STL.